pkg/controller/storagecluster/storagecluster.go

```diff
@@ -1028,11 +1029,45 @@ func (c *Controller) newSimulationPod(
 		}
 	}
 
+	// Add constraints that are used for migration. We should remove this whenever we
+	// remove the migration code
+	addMigrationConstraints(&newPod.Spec)
+
 	// Add default tolerations for StorageCluster pods
 	k8s.AddOrUpdateStoragePodTolerations(&newPod.Spec)
 	return newPod, nil
 }
 
+func addMigrationConstraints(podSpec *v1.PodSpec) {
+	if podSpec.Affinity == nil || podSpec.Affinity.NodeAffinity == nil {
+		podSpec.Affinity = &v1.Affinity{
```

Comment:
we probably only has node affinity now, but would be good to handle it

Reply:
Did not add it because we just have node affinity right now. But I will change this for future proofing, just in case we add more to placement constraints.